pkg/staticnetworkconfig/generator_test.go

```diff
@@ -21,272 +22,255 @@ func TestStaticNetworkConfig(t *testing.T) {
 
 var _ = Describe("generateConfiguration", func() {
 	var (
-		staticNetworkGenerator = StaticNetworkConfigGenerator{log: logrus.New(), nmstate: nmstate.New()}
+		staticNetworkGenerator = snc.New(logrus.New())
 	)
 
 	It("Fail with an empty host YAML", func() {
-		_, err := staticNetworkGenerator.generateConfiguration("")
+		_, err := staticNetworkGenerator.GenerateStaticNetworkConfigData(context.Background(), `[
+    {
+        "network_yaml": ""
+    }
+]`)
 		Expect(err).To(HaveOccurred())
 	})
 
 	It("Fail with an invalid host YAML", func() {
-		_, err := staticNetworkGenerator.generateConfiguration("interfaces:\n    - foo: badConfig")
+		_, err := staticNetworkGenerator.GenerateStaticNetworkConfigData(context.Background(), `[
+    {
+        "network_yaml": "interfaces:\n    - foo: badConfig"
+    }
+]`)
 		Expect(err).To(HaveOccurred())
 		Expect(err.Error()).To(ContainSubstring("InvalidArgument"))
 	})
 
 	It("Success", func() {
-		hostYaml := `interfaces:
-- ipv4:
-    address:
-    - ip: 192.0.2.1
-      prefix-length: 24
-    dhcp: false
-    enabled: true
-  name: eth0
-  state: up
-  type: ethernet`
-		config, err := staticNetworkGenerator.generateConfiguration(hostYaml)
+		var (
+			hostsYAMLS = `[
+		{
+			"network_yaml": "%s"
+		}
+	]`
+			hostYaml = `interfaces:\n- ipv4:\n    address:\n      - ip: 192.0.2.1\n        prefix-length: 24\n    dhcp: false\n    enabled: true\n  name: eth0\n  state: up\n  type: ethernet`
```

Comment:
Could this be in a more readable format?

Reply:
@rccrdpccl AFAIK,  YAML content must be properly escaped to fit within a JSON string. Newlines in hostYAML can disrupt the JSON structure. Using a string literal with \n for newlines simplifies this by avoiding manual escaping and string manipulation. I didn't find any other simple alternatives.

Reply:
Can we programmatically escape them? It's a minor concern, but if we could I think it'd make the test more readable.

Reply:
@rccrdpccl I tried something and pushed the changes. What do you think?